### PR TITLE
Properly do close code and reason

### LINKED ISSUE #2: Emitting close code as exception
Is there any reason why close codes >= 2000 get emitted as exception? The websocket specification defines close codes 4000-4999 for application-defined use, and other codes between 2000 and 3999 (inclusive) as reserved for specific uses (websocket extension, etc.).

https://github.com/voryx/WebSocketMiddleware/blob/f0a06bc75de33b09fce044621c2f70e2f8ad4af8/src/WebSocketConnection.php#L42-L47

### PR DESCRIPTION
**Pull request by ghost**
Properly parses the close code and reason from the close frame and removes the close exception (close codes >= 2000 are by definition NOT errors). This is #3, but with just the close code update. Fixes #2.

### CODE REVIEW

**Comment by ghost** [comment]
@mbonneau ping

File: src/WebSocketConnection.php [hunk 1]
```diff
@@ -32,22 +32,17 @@ function (Frame $frame) {
...
-                            $this->emit('error', [$exception, $this]);
-                            return;
+                        if ($frame->getPayloadLength() > 2) {
+                            $reason = substr($frame->getPayload(), 2);
                         }
 
-                        $this->emit('close', [$closeCode, $this]);
+                        $this->stream->end($frame->getContents());
 
+                        $this->emit('close', [$closeCode, $reason, $this]);
```
Comment (mbonneau) [inline src/WebSocketConnection.php]: This close code emission is a breaking change. I do like sending the reason back, is there a way to do this without breaking this?

**Comment by mbonneau** [comment]
@CharlotteDunois This looks good - thanks for the PR, I just had the one comment on the close emit above. Let me know your thoughts and we can work to get this merged.

File: src/WebSocketConnection.php (hunk 1 above)
Comment (ghost) [reply src/WebSocketConnection.php]: We could just put the reason as the 3rd element and keep `$this` as the 2nd element. Existing code would continue to work then.

Comment (mbonneau) [reply src/WebSocketConnection.php]: That sounds good to me. I you want to make that change I can get this merged.

Comment (ghost) [reply src/WebSocketConnection.php]: @mbonneau Changes are pushed.

**Comment by mbonneau** [comment]
@CharlotteDunois Sorry about the delay in merging, ran into a small issue with the autobahn test suite when I was trying to test this.

**Review by mbonneau — APPROVED** [review]

**Pull request merged**